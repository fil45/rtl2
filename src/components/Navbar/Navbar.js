import React from "react";
import { NavLink } from "react-router-dom";
import logo from "../../assets/octopus.png";

export default function Navbar() {
  return (
    <div className="navbar">
      <div className="brand">
        <NavLink to="/">
          <img src={logo} width="25" alt="logo" />
          <span>REACT TESTING LIBRARY DEMO</span>
        </NavLink>
      </div>
      <nav>
        <NavLink to="/users">Users</NavLink>
        <NavLink to="/redux">Redux</NavLink>
        <NavLink to="/context">Context</NavLink>
      </nav>
    </div>
  );
}
