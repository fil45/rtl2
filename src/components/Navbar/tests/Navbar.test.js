import React from "react";
import Navbar from "../Navbar";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const testedComponent = (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

describe("Navbar component --->", () => {
  it("should render the page", () => {
    const { getByText, getByAltText } = render(testedComponent);

    const logo = getByAltText("logo");
    expect(logo).toBeInTheDocument();

    const title = getByText(/REACT TESTING LIBRARY DEMO/i);
    expect(title).toBeInTheDocument();

    const linkToUsers = getByText(/Users/i);
    expect(linkToUsers).toBeInTheDocument();

    const linkToRedux = getByText(/Redux/i);
    expect(linkToRedux).toBeInTheDocument();

    const linkToContext = getByText(/Context/i);
    expect(linkToContext).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(testedComponent);
    expect(asFragment(testedComponent)).toMatchSnapshot();
  });
});
