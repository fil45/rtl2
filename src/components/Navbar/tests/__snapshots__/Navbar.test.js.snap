// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component ---> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="brand"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <img
          alt="logo"
          src="octopus.png"
          width="25"
        />
        <span>
          REACT TESTING LIBRARY DEMO
        </span>
      </a>
    </div>
    <nav>
      <a
        href="/users"
      >
        Users
      </a>
      <a
        href="/redux"
      >
        Redux
      </a>
      <a
        href="/context"
      >
        Context
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
